### PR TITLE
Added more SameExecutorCompletionStage tests.

### PR DESCRIPTION
**Pull request by billoneil**

### CODE REVIEW

File: src/main/java/org/jooq/lambda/concurrent/SameExecutorCompletionStage.java [hunk 1]
```diff
@@ -28,6 +28,9 @@
 
     @Override
     public final <U> CompletionStage<U> thenApplyAsync(Function<? super T, ? extends U> fn) {
+        if (defaultExecutor == null) {
+            return of(delegate.thenApplyAsync(fn), null);
+        }
```
Comment (billoneil) [inline src/main/java/org/jooq/lambda/concurrent/SameExecutorCompletionStage.java]: Was this what you had in mind? I think this should work. We only need to use this logic on the methods where we were supplying the default `Executor`

File: src/test/java/org/jooq/lambda/concurrent/SameExecutorCompletionStageTest.java [hunk 2]
```diff
@@ -70,8 +73,228 @@ public void testThenAccept() throws InterruptedException {
...
+        assertHits(executorA, 2);
+        assertHits(executorB, 1);
+    }
+    
+    @Test
+    public void testExceptionally() throws InterruptedException {
+        CompletionStage<Void> firstCompletionStage = Async.supplyAsync(latchSupplier(latch1), executorA);
+        firstCompletionStage = firstCompletionStage.exceptionally(ex -> null);
+        latch1.await();
+        assertHits(executorA, 1);
```
Comment (billoneil) [inline src/test/java/org/jooq/lambda/concurrent/SameExecutorCompletionStageTest.java]: I'm not sure if this test is doing anything useful.

File: src/main/java/org/jooq/lambda/concurrent/SameExecutorCompletionStage.java (hunk 1 above)
Comment (lukaseder) [inline src/main/java/org/jooq/lambda/concurrent/SameExecutorCompletionStage.java]: Yes, that's what I had in mind

**Comment by lukaseder** [comment]
Looks good to me, thank you very much, again!

**Pull request merged**